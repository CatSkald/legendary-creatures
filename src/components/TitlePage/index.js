import styles from "./index.module.scss";

import React from "react";
import PropTypes from "prop-types";

const TitlePage = (props) => {
  return <h1 className={styles.title}>{props.text}</h1>;
};

TitlePage.propTypes = {
  text: PropTypes.string.isRequired,
};

export default TitlePage;
