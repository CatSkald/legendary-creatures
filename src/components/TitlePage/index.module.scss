.title {
  line-height: 140%;
  font-size: 3rem;
  font-weight: bold;
  margin-bottom: var(--space-lg);

  @media screen and (min-width: 768px /*--screen-medium*/) {
    line-height: 1.1;
    font-size: 3.5rem;
  }

  @media screen and (min-width: 1056px /*--screen-large*/) {
    line-height: 1.1;
    font-size: 4rem;
  }
}
