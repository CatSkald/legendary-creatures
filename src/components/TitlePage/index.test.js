import React from "react";
import { render } from "@testing-library/react";

import TitlePage from "../TitlePage";

describe("TitlePage", () => {
  it("renders correctly", () => {
    const { container } = render(<TitlePage text="Hello, World" />);
    expect(container).toMatchSnapshot();
  });
});
