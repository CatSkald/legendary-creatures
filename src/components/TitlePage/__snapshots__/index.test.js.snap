// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitlePage renders correctly 1`] = `
<div>
  <h1
    class="TitleElement"
  >
    Hello, World
  </h1>
</div>
`;
